plugins/CMakeLists.txt

```diff
@@ -12,85 +11,69 @@ if(OBS_CMAKE_VERSION VERSION_GREATER_EQUAL 3.0.0)
   set_property(GLOBAL APPEND PROPERTY OBS_FEATURES_ENABLED "Plugin Support")
 
   macro(check_obs_browser)
-    if(NOT (OS_WINDOWS AND NOT CMAKE_GENERATOR_PLATFORM MATCHES "(Win32|x64)"))
-      if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-browser/cmake/macos/Info.plist.in")
-        add_subdirectory(obs-browser)
-      elseif(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-browser/CMakeLists.txt")
+    if((OS_WINDOWS AND CMAKE_GENERATOR_PLATFORM MATCHES "(Win32|x64)")
+       OR OS_MACOS
+       OR OS_LINUX)
+      if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-browser/CMakeLists.txt")
         message(FATAL_ERROR "Required submodule 'obs-browser' not available.")
+      else()
+        add_subdirectory(obs-browser)
       endif()
+    else()
+      add_custom_target(obs-browser)
+      target_disable(obs-browser)
     endif()
   endmacro()
 
   macro(check_obs_websocket)
-    if(OS_WINDOWS
-       OR OS_MACOS
-       OR OS_LINUX)
-      if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-websocket/cmake/macos/Info.plist.in")
-        add_subdirectory(obs-websocket)
-      elseif(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-websocket/CMakeLists.txt")
-        message(FATAL_ERROR "Required submodule 'obs-websocket' not available.")
-      endif()
+    if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-websocket/CMakeLists.txt")
+      message(FATAL_ERROR "Required submodule 'obs-websocket' not available.")
+    else()
+      add_subdirectory(obs-websocket)
     endif()
   endmacro()
 
   # Add plugins in alphabetical order to retain order in IDE projects
-  add_subdirectory(aja)
-  if(OS_WINDOWS OR OS_MACOS)
-    add_subdirectory(coreaudio-encoder)
-  endif()
-  if(OS_WINDOWS
-     OR OS_MACOS
-     OR OS_LINUX)
-    add_subdirectory(decklink)
-  endif()
-  add_subdirectory(image-source)
-  if(OS_LINUX)
-    add_subdirectory(linux-alsa)
-    add_subdirectory(linux-capture)
-    add_subdirectory(linux-jack)
-    add_subdirectory(linux-pipewire)
-    add_subdirectory(linux-pulseaudio)
-    add_subdirectory(linux-v4l2)
-  endif()
-  if(OS_MACOS)
-    add_subdirectory(mac-avcapture)
-    add_subdirectory(mac-capture)
-    add_subdirectory(mac-syphon)
-    add_subdirectory(mac-videotoolbox)
-    add_subdirectory(mac-virtualcam)
-  endif()
+  add_obs_plugin(aja PLATFORMS WINDOWS MACOS LINUX WITH_MESSAGE)
```

Comment:
```
/build/obs-studio-git/src/obs-studio/plugins/aja/aja-routing.cpp: In static member function ‘static void aja::Routing::StartSourceAudio(const SourceProps&, CNTV2Card*)’:
/build/obs-studio-git/src/obs-studio/plugins/aja/aja-routing.cpp:150:31: error: ‘virtual bool CNTV2Card::WriteAudioSource(ULWord, NTV2Channel)’ is deprecated [-Werror=deprecated-declarations]
  150 |         card->WriteAudioSource(0, channel);
      |         ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
In file included from /usr/include/ajalibraries/ajantv2/includes/ntv2publicinterface.h:11,
                 from /usr/include/ajalibraries/ajantv2/includes/ntv2vpid.h:12,
                 from /build/obs-studio-git/src/obs-studio/plugins/aja/aja-enums.hpp:3,
                 from /build/obs-studio-git/src/obs-studio/plugins/aja/aja-props.hpp:3,
                 from /build/obs-studio-git/src/obs-studio/plugins/aja/aja-card-manager.hpp:3,
                 from /build/obs-studio-git/src/obs-studio/plugins/aja/aja-routing.cpp:1:
/usr/include/ajalibraries/ajantv2/includes/ntv2card.h:2482:21: note: declared here
 2482 |         AJA_VIRTUAL NTV2_DEPRECATED_f(bool WriteAudioSource (const ULWord inValue, const NTV2Channel inChannel = NTV2_CHANNEL1));       ///< @deprecated        This function is obsolete.
      |                     ^~~~~~~~~~~~~~~~~
/build/obs-studio-git/src/obs-studio/plugins/aja/aja-routing.cpp:185:34: error: ‘virtual bool CNTV2Card::ReadAudioSource(ULWord&, NTV2Channel)’ is deprecated [-Werror=deprecated-declarations]
  185 |         if (card->ReadAudioSource(oldValue, channel)) {
      |             ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
/usr/include/ajalibraries/ajantv2/includes/ntv2card.h:2483:21: note: declared here
 2483 |         AJA_VIRTUAL NTV2_DEPRECATED_f(bool ReadAudioSource (ULWord & outValue, const NTV2Channel inChannel = NTV2_CHANNEL1));   ///< @deprecated        This function is obsolete.
      |                     ^~~~~~~~~~~~~~~~~
/build/obs-studio-git/src/obs-studio/plugins/aja/aja-routing.cpp:186:39: error: ‘virtual bool CNTV2Card::WriteAudioSource(ULWord, NTV2Channel)’ is deprecated [-Werror=deprecated-declarations]
  186 |                 card->WriteAudioSource(oldValue | magicAudioBits, channel);
      |                 ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/ajalibraries/ajantv2/includes/ntv2card.h:2482:21: note: declared here
 2482 |         AJA_VIRTUAL NTV2_DEPRECATED_f(bool WriteAudioSource (const ULWord inValue, const NTV2Channel inChannel = NTV2_CHANNEL1));       ///< @deprecated        This function is obsolete.
      |                     ^~~~~~~~~~~~~~~~~
/build/obs-studio-git/src/obs-studio/plugins/aja/aja-routing.cpp: In static member function ‘static void aja::Routing::ConfigureOutputAudio(const OutputProps&, CNTV2Card*)’:
/build/obs-studio-git/src/obs-studio/plugins/aja/aja-routing.cpp:543:42: error: ‘virtual bool CNTV2Card::SetAudioOutputMonitorSource(NTV2AudioMonitorSelect, NTV2Channel)’ is deprecated [-Werror=deprecated-declarations]
  543 |         card->SetAudioOutputMonitorSource(NTV2_AudioChannel1_2, channel);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/ajalibraries/ajantv2/includes/ntv2card.h:7406:28: note: declared here
 7406 |         AJA_VIRTUAL inline NTV2_DEPRECATED_f(bool SetAudioOutputMonitorSource (const NTV2AudioMonitorSelect inChannelPair, const NTV2Channel inAudioSystem = NTV2_CHANNEL1))    {return SetAudioOutputMonitorSource(inChannelPair, NTV2AudioSystem(inAudioSystem));}    ///< @deprecated        Use the function that uses NTV2AudioChannelPair and NTV2AudioSystem params.
      |                            ^~~~~~~~~~~~~~~~~
```

Regressions were found with `-Wno-error=unused-variable` to be able to test further